pkg/kubectl/cmd/util/editor/editoptions.go

```diff
@@ -98,9 +108,46 @@ func NewEditOptions(editMode EditMode, ioStreams genericclioptions.IOStreams) *E
 }
 
 type editPrinterOptions struct {
-	printer   printers.ResourcePrinter
-	ext       string
-	addHeader bool
+	printFlags *genericclioptions.PrintFlags
+	ext        string
+	addHeader  bool
+}
+
+func (e *editPrinterOptions) Complete(fromPrintFlags *genericclioptions.PrintFlags) error {
+	if e.printFlags == nil {
+		return fmt.Errorf("missing PrintFlags in editor printer options")
+	}
+
+	// bind output format from existing printflags
+	if fromPrintFlags != nil {
+		e.printFlags.OutputFormat = fromPrintFlags.OutputFormat
+	}
+
+	// default to yaml only when existing output is empty
+	defaultOutputFormat := "yaml"
+	if len(*e.printFlags.OutputFormat) == 0 {
+		e.printFlags.OutputFormat = &defaultOutputFormat
+	}
+
+	if *e.printFlags.OutputFormat == "json" {
```

Comment:
I would like to avoid doing this everywhere if possible, but in this case, we need to have the command know about the PrintFlags.OutputFormat field to complete header / file ext info

Reply:
> I would like to avoid doing this everywhere if possible, but in this case, we need to have the command know about the PrintFlags.OutputFormat field to complete header / file ext info

I don't remember this.  What header?  As long as the ugly is contained I don't care too much, but this looks really weird.

Reply:
we print out a commented header at the top of the file, when editing an object with the user's default editor